pkg/virt-config/virt-config.go

```diff
@@ -209,11 +209,31 @@ func (c *ClusterConfig) GetPermittedHostDevices() *v1.PermittedHostDevices {
 	return c.GetConfig().PermittedHostDevices
 }
 
-func (c *ClusterConfig) GetDesiredMDEVTypes(nodeName string) []string {
+func (c *ClusterConfig) GetDesiredMDEVTypes(node *k8sv1.Node) []string {
 	mdevTypesConf := c.GetConfig().MediatedDevicesConfiguration
 	if mdevTypesConf == nil {
 		return []string{}
 	}
+	nodeMdevConf := mdevTypesConf.MediatedDevicesTypes.NodeMediatedDeviceTypesConfig
+	if nodeMdevConf != nil {
+		mdevTypesMap := make(map[string]struct{})
+		for _, nodeConfig := range nodeMdevConf {
+			for key, _ := range nodeConfig.NodeSelector {
+				if _, exist := node.Labels[key]; exist {
+					for _, mdevType := range nodeMdevConf.MediatedDevicesTypes {
+						mdevTypesMap[mdevType] = struct{}{}
+					}
```

Comment:
you likely don't need to run this loop for every label. instead just once when the match is made. running it multiple times doesn't hurt anything, but isn't necessary, so it's a small optimization.